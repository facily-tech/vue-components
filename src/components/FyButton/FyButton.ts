/// <reference path="../../../node_modules/vuetify/src/globals.d.ts" />
import Vue from 'vue';

import { VBtn } from 'vuetify/lib';

const base = Vue.extend({ mixins: [VBtn] });

type options = InstanceType<typeof base>;

export default base.extend<options>().extend({
  name: 'fy-button',

  props: {
    elevation: {
      type: Number,
      default: 0,
    },
  },

  computed: {
    classes(): Record<string, unknown> {
      return {
        ...VBtn.options.computed.classes.call(this),
        'fy-button': true,
      };
    },
  },
});
