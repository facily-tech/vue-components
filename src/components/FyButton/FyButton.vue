<template>
  <v-btn :elevation="0" v-bind="$attrs" v-on="$listeners">
    <slot v-for="(_, name) in $slots" :name="name" :slot="name" />
  </v-btn>
</template>

<script lang="ts">
import Vue from 'vue';

export default Vue.extend({
  name: 'FyButton',
});
</script>
