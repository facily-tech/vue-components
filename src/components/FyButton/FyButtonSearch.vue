<template>
  <fy-button
    outlined
    :color="color"
    class="fy-button-search"
    :data-test="dataTest"
    @click="$emit('click-filter')"
    v-bind="propsAndAttrs"
    v-on="$listeners"
  >
    <v-icon>mdi-magnify</v-icon>
    <span class="text-uppercase" :class="isHideName">
      {{ name }}
    </span>
  </fy-button>
</template>

<script lang="ts">
import Vue from 'vue';
import FyButton from './FyButton.vue';

export default Vue.extend({
  name: 'FyButtonSearch',

  components: {
    FyButton,
  },

  props: {
    name: {
      type: String,
      default: 'Buscar',
    },

    hideName: {
      type: Boolean,
      default: false,
    },

    dataTest: {
      type: String,
      default: 'btn-filter',
    },
  },

  computed: {
    propsAndAttrs(): Record<string, unknown> {
      return {
        ...this.$attrs,
        ...this.$props,
      };
    },

    isHideName() {
      return this.hideName ? 'd-none d-sm-flex' : '';
    },

    color(): string {
      return this.$vuetify?.theme?.dark ? 'grey' : 'primary';
    }
  },
});
</script>
