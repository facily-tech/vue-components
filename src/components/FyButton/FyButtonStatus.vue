<template>
  <fy-button
    :color="color"
    @click="$emit('active-control', { key: event, value: item })"
    data-test="btn-status"
    x-small
    text
    v-bind="propsAndAttrs"
    v-on="$listeners"
  >
    <span class="text-uppercase">{{ name }}</span>
    <v-icon small class="pl-1">mdi-circle</v-icon>
  </fy-button>
</template>

<script lang="ts">
import Vue from 'vue';
import FyButton from './FyButton.vue';

export default Vue.extend({
  name: 'FyButtonStatus',

  components: {
    FyButton,
  },

  props: {
    item: {
      type: Object,
    },

    status: {
      type: Boolean,
      default: true,
    },
  },

  computed: {
    propsAndAttrs(): Record<string, unknown> {
      return {
        ...this.$attrs,
        ...this.$props,
      };
    },

    name() {
      return this.status ? 'Ativo' : 'Inativo';
    },

    color() {
      return this.status ? 'success' : 'warning';
    },

    event() {
      return !this.status ? 'actitve' : 'inactitve';
    },
  },
});
</script>
