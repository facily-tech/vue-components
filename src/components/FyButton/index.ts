import FyButton from './FyButton.vue';
export { default as FyButtonSearch } from './FyButtonSearch.vue';

export { FyButton };
export default FyButton;
