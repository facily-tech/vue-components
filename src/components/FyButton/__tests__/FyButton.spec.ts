import Vue from 'vue';
import Router from 'vue-router';
import FyButton from '../FyButton.vue';
import { createLocalVue, mount, Wrapper } from '@vue/test-utils';
import Vuetify from 'vuetify';

Vue.use(Vuetify);

describe('FyButton', () => {
	// eslint-disable-next-line
  let mountFunction: (options?: Record<string, unknown>) => Wrapper<any>;
  let router: Router;
  let localVue: typeof Vue;

  beforeEach(() => {
    router = new Router();
    localVue = createLocalVue();
    localVue.use(Router);

    mountFunction = (options = {}) => {
      return mount(FyButton, {
        localVue,
        vuetify: new Vuetify(),
        router,
        ...options,
      });
    };
  });

  it('should render component and match snapshot', () => {
    expect(mountFunction().html()).toMatchSnapshot();
  });
});
