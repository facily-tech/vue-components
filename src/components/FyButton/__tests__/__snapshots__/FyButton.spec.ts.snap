// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FyButton should render component and match snapshot 1`] = `<button type="button" class="v-btn v-btn--has-bg theme--light elevation-0 v-size--default" style="min-height: 40px;" data-test="btn"><span class="v-btn__content"></span></button>`;
